src/rules/orderedImportsRule.ts

```diff
@@ -127,64 +170,105 @@ export class Rule extends Lint.Rules.AbstractRule {
 type Transform = (x: string) => string;
 const TRANSFORMS = new Map<string, Transform>([
     ["any", () => ""],
-    ["case-insensitive", (x) => x.toLowerCase()],
+    ["case-insensitive", x => x.toLowerCase()],
     ["lowercase-first", flipCase],
-    ["lowercase-last", (x) => x],
-    ["full", (x) => x],
-    ["basename", (x) => {
-        if (!ts.isExternalModuleNameRelative(x)) {
-            return x;
-        }
+    ["lowercase-last", x => x],
+    ["full", x => x],
+    [
+        "basename",
+        x => {
+            if (!ts.isExternalModuleNameRelative(x)) {
+                return x;
+            }
 
-        const splitIndex = x.lastIndexOf("/");
-        if (splitIndex === -1) {
-            return x;
+            const splitIndex = x.lastIndexOf("/");
+            if (splitIndex === -1) {
+                return x;
+            }
+            return x.substr(splitIndex + 1);
         }
-        return x.substr(splitIndex + 1);
-    }],
+    ]
 ]);
 
 enum ImportType {
     LIBRARY_IMPORT = 1,
     PARENT_DIRECTORY_IMPORT = 2, // starts with "../"
-    CURRENT_DIRECTORY_IMPORT = 3, // starts with "./"
+    CURRENT_DIRECTORY_IMPORT = 3 // starts with "./"
 }
 
 interface Options {
     groupedImports: boolean;
     importSourcesOrderTransform: Transform;
-    moduleSourcePath: Transform;
+    moduleSourceTransform: Transform;
     namedImportsOrderTransform: Transform;
+    groups: GroupOption[];
+}
+
+interface GroupOption {
+    name: string;
+    match: RegExp | null;
+    order: number;
 }
 
 interface JsonOptions {
     "grouped-imports"?: boolean;
     "import-sources-order"?: string;
     "named-imports-order"?: string;
     "module-source-path"?: string;
+    groups?: JsonGroupOption[];
+}
+
+interface JsonGroupOption {
+    name?: string;
+    match: string | null;
+    order: number;
 }
 
 function parseOptions(ruleArguments: any[]): Options {
     const optionSet = (ruleArguments as JsonOptions[])[0];
+
+    const defaultGroups: JsonGroupOption[] = [
+        { name: "parent directories", match: "^[.][.]", order: 20 },
+        { name: "current directory", match: "^[.]", order: 30 },
```

Comment:
prefer `\.` over `[.]` - more obvious

Reply:
I agree but as far as I can tell that isn't allowed in TS/JS inside a string passed to the RegExp.

For example:

```
> RegExp('^\.').test('foo')
> true
> RegExp('^[.]').test('foo')
> false
```
Looks like Javascript will process the "\." escape character when assigning the string and just turn it into "." before it get's to the regex.

Other tools I saw that tried to do similar sorting ended up configuring it with [.][.] as well.

Reply:
ah good point on the string. how about `\\.`? i guess that's the same length, but `[.]` relies on the weird truth that `.` in a character class is treated as the literal `.` character, not the "match-anything" that it means in other contexts.